runners/flink/src/main/java/org/apache/beam/runners/flink/translation/utils/FlinkPipelineTranslatorUtils.java

```diff
@@ -77,4 +77,25 @@ public static WindowingStrategy getWindowingStrategy(
           e);
     }
   }
+
+  public static String genOperatorNameFromStagePayload(
+      RunnerApi.ExecutableStagePayload stagePayload) {
+    StringBuilder sb = new StringBuilder();
+    final int transformsCount = stagePayload.getTransformsCount();
+    sb.append("[").append(transformsCount).append("]");
+    sb.append("{");
+    for (int i = 0; i < transformsCount; i++) {
+      String name = stagePayload.getTransforms(i);
+      // Python: Remove the 'ref_AppliedPTransform_' prefix which just makes the name longer
```

Comment:
It would probably be better to use stagePayload.components.transforms[name].unique_name rather than its id here.

Reply:
Good suggestion. The unique name doesn't include the id but makes it more readable: 

```
// Master
beam:env:docker:v1:1
// Old PR version
[5]{read/Read/Reshuffle/ReshufflePerKey/FlatMap(restore_timestamps)_14, read/Read/Reshuffle/RemoveRandomKeys_15, read/Read/ReadSplits_16, split_17, pair_with_one_18}
// Using unique_name
[5]{read/Read/Reshuffle/ReshufflePerKey/FlatMap(restore_timestamps), read/Read/Reshuffle/RemoveRandomKeys, read/Read/ReadSplits, split, pair_with_one}
```